File: services/orders-service/src/main/java/pl/xsware/orders/application/event/PaymentRequestedEvent.java
package pl.xsware.orders.application.event;

import pl.xsware.orders.domain.order.Currency;
import pl.xsware.orders.domain.shared.OutboxEvent;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

public record PaymentRequestedEvent(
    UUID eventId,
    String eventType,
    int version,
    Instant occurredAt,
    Data data
) implements OutboxEvent {

    public static final String TYPE = "PaymentRequested";
    public static final int VERSION = 1;

    public static PaymentRequestedEvent of(
        UUID orderId,
        BigDecimal amount,
        Currency currency
    ) {
        return new PaymentRequestedEvent(
            UUID.randomUUID(),
            TYPE,
            VERSION,
            Instant.now(),
            new Data(orderId, amount, currency)
        );
    }

    @Override
    public String aggregateType() {
        return "ORDER";
    }

    @Override
    public String aggregateId() {
        return data.orderId().toString();
    }

    public record Data(
        UUID orderId,
        BigDecimal amount,
        Currency currency
    ) {}
}

File: services/payments-service/src/main/kotlin/pl/xsware/payments/application/event/PaymentRequestedEvent.kt
package pl.xsware.payments.application.event

import java.time.Instant
import java.util.UUID

data class PaymentRequestedEvent(
    val eventId: UUID,
    val eventType: String,
    val version: Int,
    val occurredAt: Instant,
    val data: Data
) {

    companion object {
        const val TYPE = "PaymentRequested"
        const val VERSION = 1
    }

    data class Data(
        val orderId: UUID,
        val amount: String,
        val currency: String
    )
}
